workers/worker.py

```
@@ -45,6 +50,189 @@
+    """
+
+    requires = {"celery.worker.components:Pool"}
+
+    def __init__(self, worker, worker_type, **kwargs):
+        self.worker = worker
+        self.worker_type = worker_type
+        self.heartbeat_thread = None
+        self.should_stop = threading.Event()
+        self._shutdown_in_progress = False
+
+    def start(self, worker):
```

Ruff (0.14.1)

Unused method argument: `worker`

(ARG002)